src/myopic.jl

```diff
@@ -161,81 +154,76 @@ end
 _shape_epsps(e_temp, sim, ::Myopic) = e_temp
 _shape_epsps(e_temp, sim, ::MyopicFast) = repeat(e_temp, m_out(sim)÷length(sim.tsteps.dts))
 
-function _entropy(model::BinomialGridModel, obs::BinomialObservation, ::Myopic)
+function _entropy(model::BinomialGridModel, obs::BinomialObservation, policy::Myopic)
     # CPU algorithm: move index arrays to CPU
     Nind = Array(model.Nind)
     pind = Array(model.pind)
     qind = Array(model.qind)
     σind = Array(model.σind)
     τind = Array(model.τind)
 
+    Nrng = Array(model.Nrng)
+    prng = Array(model.prng)
+    qrng = Array(model.qrng)
+    σrng = Array(model.σrng)
+    τrng = Array(model.τrng)
+
     dts = Array(obs.dt)
 
+    η = policy.penalty
+
     minent = Inf
     imin = 0
     @inbounds for i in 1:size(Nind, 1)
-        dict = Dict{NTuple{5, Int64}, Int}()
-        @inbounds for j in 1:size(Nind, 2)
-            iN = Nind[i, j]
-            ip = pind[i, j]
-            iq = qind[i, j]
-            iσ = σind[i, j]
-            iτ = τind[i, j]
-            key = (iN, ip, iq, iσ, iτ)
-            dict[key] = get!(dict, key, 0) + 1
-        end
-        ent = 0.
-        for value in values(dict)
-            p = value/size(Nind, 2)
-            ent -= p * log(p)
-        end
-        if ent < minent
-            minent = ent
+        samples = [Nrng[Nind[i, :]]';prng[pind[i, :]]';qrng[qind[i, :]]';σrng[σind[i, :]]';τrng[τind[i, :]]']
+        Σ_est = cov(samples')
+
+        ent = 0.5*log(det(2*pi*ℯ*Σ_est))
+
+        if ent + η*dts[i] < minent
+            minent = ent + η*dts[i]
             imin = i 
         end
     end
     return dts[imin]
 end
```

Comment:
- Keep the old function and define a new one `_diffentropy_approx`
- Do this directly on the device holding the parameter arrays (GPU if on GPU already):
```julia
N = model.N
p = model.p
# etc.
@inbounds for i in 1:size(Nind, 1)
    @views M = hcat(N[i, :], p[i, :], q[i, :], σ[i, :], τ[i, :]) # use @views to not allocate additional memory
    Σ_est = cov(M)
end
# etc.
```